src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketMirrorHandler.java

```diff
@@ -0,0 +1,122 @@
+    }
+
+    public StandardListBoxModel fetchAsListBox(MirrorFetchRequest mirrorFetchRequest) {
+        StandardListBoxModel options = new StandardListBoxModel();
+        options.add(new Option(DEFAULT_UPSTREAM_SERVER, ""));
+        if (isEmpty(mirrorFetchRequest.getServerId()) ||
+            isEmpty(mirrorFetchRequest.getProjectNameOrKey()) ||
+            isEmpty(mirrorFetchRequest.getRepoNameOrSlug())) {
+            return options;
+        }
+
+        String existingSelection = mirrorFetchRequest.getExistingMirrorSelection();
+        fetchRepositoriesQuietly(mirrorFetchRequest)
+                .stream()
+                .map(mirroredRepo -> createOption(existingSelection, mirroredRepo))
+                .forEach(option -> options.add(option));
+
+        return options;
+    }
+
+    private List<EnrichedBitbucketMirroredRepository> fetchRepositores(MirrorFetchRequest mirrorFetchRequest) {
+        BitbucketServerConfiguration serverConfiguration =
+                bitbucketPluginConfiguration.getServerById(mirrorFetchRequest.getServerId())
+                        .orElseThrow(() -> new MirrorFetchException("Server config not found"));
+        String bitbucketBaseUrl = requireNonNull(serverConfiguration.getBaseUrl(), "Bitbucket base Url not found");
+
+        BitbucketCredentials jobOrGlobalConf =
+                jenkinsToBitbucketCredentials.toBitbucketCredentials(mirrorFetchRequest.getJobCredentials(), serverConfiguration);
+        BitbucketClientFactory client = bitbucketClientFactoryProvider.getClient(bitbucketBaseUrl, jobOrGlobalConf);
+        BitbucketRepository repository =
+                bitbucketRepoFetcher.fetchRepo(client, mirrorFetchRequest.getProjectNameOrKey(), mirrorFetchRequest.getRepoNameOrSlug());
+        BitbucketMirrorClient mirrorClient = client.getMirroredRepositoriesClient(repository.getId());
+        return mirrorClient.getMirroredRepositoryDescriptors()
+                .getValues()
+                .stream()
+                .map(repoDescriptor -> fetchMirroredRepo(mirrorClient, repoDescriptor, repository.getId()))
+                .filter(BitbucketMirroredRepository::isAvailable)
+                .map(mirrorDetails -> new EnrichedBitbucketMirroredRepository(repository, mirrorDetails))
+                .collect(Collectors.toList());
+    }
+
+    private Option createOption(String existingSelection,
+                                EnrichedBitbucketMirroredRepository mirroredRepo) {
+        String mirrorName = mirroredRepo.getMirroringDetails().getMirrorName();
+        return new Option(mirrorName, mirrorName, mirrorName.equals(existingSelection));
+    }
+
+    private List<EnrichedBitbucketMirroredRepository> fetchRepositoriesQuietly(MirrorFetchRequest mirrorFetchRequest) {
+        try {
+            return fetchRepositores(mirrorFetchRequest);
+        } catch (BitbucketClientException ex) {
```

Comment:
Because the `MirrorFetchException` does not extend the `BitbucketClientException` it will bubble out of here. I think we should probably change the `MirrorFetchException` to extend our client exception?

Reply:
Fixed